juju/juju
# refactor: remove set status support for environ upgrade

In this PR we are removing the ability to set a model's status from within the environ upgrade workers for IAAS and CAAS. The reason we are doing this is model status is now computed under the domain implementation and we are dropping support for environ upgrading in 4.0.

This is one of a few tasks to slowly weave out environ upgrading. The first step is breaking the mongo dependencies so we can cut model status over.

## Checklist


## QA steps

We want to make sure that the environ upgrade gates don't get stuck and pause the manifold.

1. Bootstrap a new controller on both LXD and K8s
2. Add a new model
3. Deploy something on both providers and confirm everything is happy with nothing stuck.

Don't worry about what the model status might be reporting at the moment.

## Documentation changes

N/A

## Links

**Jira card:** JUJU-7110
**Jira card:** JUJU-7109
